iroh-resolver/src/unixfs_builder.rs

```diff
@@ -325,45 +450,75 @@ pub async fn add_file<S: Store>(path: &Path, rpc: Option<S>) -> Result<Cid> {
     Ok(root.expect("missing root"))
 }
 
+#[async_recursion(?Send)]
+async fn make_dir_from_path<P: Into<PathBuf>>(path: P, recursive: bool) -> Result<Directory> {
+    let path = path.into();
+    let mut dir = DirectoryBuilder::new();
+    dir.with_name(
+        path.file_name()
+            .and_then(|s| s.to_str())
+            .unwrap_or_default(),
+    );
+    if recursive {
+        dir.allow_recursive();
+    }
+    let entries = std::fs::read_dir(path.clone())?;
```

Comment:
this should use tokios read_dir to make sure we do not block

Reply:
TYTYTY. I was wondering about this. Need to get it through my head that we are doing async read/write!!!